tests/v2/test_waku_rln_relay_onchain.nim

```diff
@@ -0,0 +1,263 @@
+    debug "The hash of registration tx: ", tx # value is the membership fee
+
+    # var members: array[2, uint256] = [20.u256, 21.u256]
+    # debug "This is the batch registration result ", await sender.registerBatch(members).send(value = (members.len * MEMBERSHIP_FEE)) # value is the membership fee
+
+    let balance = await web3.provider.eth_getBalance(web3.defaultAccount, "latest")
+    debug "Balance after registration: ", balance
+
+    await web3.close()
+    debug "disconnected from", ETH_CLIENT
+
+  asyncTest "registration procedure":
+    # preparation ------------------------------
+    # deploy the contract
+    let contractAddress = await uploadRLNContract(ETH_CLIENT)
+
+    # prepare rln-relay peer inputs
+    let
+      web3 = await newWeb3(ETH_CLIENT)
+      accounts = await web3.provider.eth_accounts()
+      # choose one of the existing accounts for the rln-relay peer
+      ethAccountAddress = accounts[9]
+    await web3.close()
+
+    # create an RLN instance
+    var rlnInstance = createRLNInstance()
+    check: rlnInstance.isOk == true
+
+    # generate the membership keys
+    let membershipKeyPair = membershipKeyGen(rlnInstance.value)
+    check: membershipKeyPair.isSome
+
+    # test ------------------------------
+    # initialize the WakuRLNRelay
+    var rlnPeer = WakuRLNRelay(membershipKeyPair: membershipKeyPair.get(),
+      membershipIndex: MembershipIndex(0),
+      ethClientAddress: ETH_CLIENT,
+      ethAccountAddress: ethAccountAddress,
+      membershipContractAddress: contractAddress)
+
+    # register the rln-relay peer to the membership contract
+    let is_successful = await rlnPeer.register()
+    check: is_successful
+
+  asyncTest "mounting waku rln-relay":
+    # preparation ------------------------------
+    let
+      nodeKey = crypto.PrivateKey.random(Secp256k1, rng[])[]
+      node = WakuNode.new(nodeKey, ValidIpAddress.init("0.0.0.0"),
+        Port(60000))
+    await node.start()
+
+    # deploy the contract
+    let membershipContractAddress = await uploadRLNContract(ETH_CLIENT)
+
+    # prepare rln-relay inputs
+    let
+      web3 = await newWeb3(ETH_CLIENT)
+      accounts = await web3.provider.eth_accounts()
+      # choose one of the existing account for the rln-relay peer
+      ethAccountAddress = accounts[9]
+    await web3.close()
+
+    # create current peer's pk
+    var rlnInstance = createRLNInstance()
+    check rlnInstance.isOk == true
+    var rln = rlnInstance.value
+    # generate a key pair
+    var keypair = rln.membershipKeyGen()
+    doAssert(keypair.isSome())
+
+    # current peer index in the Merkle tree
+    let index = uint(5)
+
+    # Create a group of 10 members
+    var group = newSeq[IDCommitment]()
+    for i in 0..10:
+      var member_is_added: bool = false
+      if (uint(i) == index):
+        #  insert the current peer's pk
+        group.add(keypair.get().idCommitment)
+        member_is_added = rln.insertMember(keypair.get().idCommitment)
```

Comment:
Is it better CamelCase?

Reply:
sure, tracking issue https://github.com/status-im/nim-waku/issues/946